src/app/ssh-keys/redux/ssh-key.reducers.ts

```diff
@@ -0,0 +1,203 @@
+import { createEntityAdapter, EntityAdapter, EntityState } from '@ngrx/entity';
+import { SSHKeyPair } from '../../shared/models/ssh-keypair.model';
+import {
+  combineReducers,
+  compose,
+  createFeatureSelector,
+  createSelector
+} from '@ngrx/store';
+import { Account } from '../../shared/models/account.model';
+
+import * as sshKey from './ssh-key.actions';
+
+export interface State {
+  list: ListState,
+  form: FormState
+}
+
+export interface ListState extends EntityState<SSHKeyPair> {
+  filters: {
+    selectedGroupings: any[],
+    selectedAccounts: Account[]
+  }
+}
+
+const initialListState: ListState = {
+  ids: [],
+  entities: null,
+  filters: {
+    selectedAccounts: [],
+    selectedGroupings: []
+  }
+};
+
+export interface FormState {
+  loading: boolean,
+  error: Object
```

Comment:
https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html